lib/rules/no-assignment-of-untracked-properties-used-in-tracking-contexts.js

```diff
@@ -0,0 +1,226 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+const types = require('../utils/types');
+const decoratorUtils = require('../utils/decorators');
+const javascriptUtils = require('../utils/javascript');
+const propertySetterUtils = require('../utils/property-setter');
+const assert = require('assert');
+const { getImportIdentifier } = require('../utils/import');
+
+const ERROR_MESSAGE =
+  "Use `set(this, 'propertyName', 'value')` instead of assignment for untracked properties that are used as computed property dependencies (or convert to using tracked properties).";
+
+/**
+ * Gets the list of string dependent keys from a computed property.
+ *
+ * @param {Node} node - the computed property node
+ * @returns {String[]} - the list of string dependent keys from this computed property
+ */
+function getComputedPropertyDependentKeys(node) {
+  if (!node.arguments) {
+    return [];
+  }
+
+  return node.arguments
+    .filter((arg) => arg.type === 'Literal' && typeof arg.value === 'string')
+    .map((node) => node.value);
```

Comment:
Do we need to do anything special for `@each` here?

Reply:
Good point, updated to handle that, and added a test case.